include/ml/classifiers/svm/smo.hpp

```diff
@@ -298,7 +298,7 @@ smo_out<T> smo(queue& q, matrix_t<T>& x, vector_t<T>& y, T c, T tol, T eps, SYCL
   assert_eq(y.kernel_range.get_global_linear_range(), to_pow2(m));
 
   if (max_nb_iter == 0)
-    max_nb_iter = std::max(10000000LU, m > INT_MAX/100 ? INT_MAX : 100*m);
+    max_nb_iter = std::max(10000000ULL, m > std::numeric_limits<size_t>::max()/100ULL ? std::numeric_limits<size_t>::max() : 100ULL*m);
```

Comment:
This will break on Unix since the index type is `unsigned long` there. I'd like to know why it is `long long` on Windows, maybe Mehdi knows more about this.
If possible I'd like to have a clean solution on the Eigen side for this, if it's not possible I'll use `decltype` instead here.

Reply:
size_t is a long long on Windows. You could do a static cast into the size_t type but that seems ugly. You should probably just make sure that all the types in that call are the same, whatever they might be. I guess m might be the problem here?

Reply:
Oh ok, that's annoying. m is part of the problem yes, I have 2 constants that should be of type size_t (for min and max type deduction). It's not great that I have these constants but that's an issue for another time, I'll have to static_cast them for now.
Thanks!